### ISSUE TITLE
Question with using GPU

### ISSUE THREAD

**Issue by ligongzheng** [luoyetx/mini-caffe#51]
Hello,
    Why the GPU is used when I didn't invoke caffe::SetMode(caffe::GPU,0) in my program ? Although I complied it with options -DUSE_CUDA=ON -DUSE_CUDNN=ON, the value of mode_  is Caffe::CPU and it should work in CPU style (I get it from common.cpp).   In my thought,even though I complied it with cuda and cudnn on, the mode is still Caffe::CPU if I didn't invoke the SetMode function and GPU should not be occupied (I check it with NVIDIA-SMI).

**Comment by luoyetx** [comment]
Yes, It's a bug. Checkout branch [`fix-create-layer`](https://github.com/luoyetx/mini-caffe/tree/fix-create-layer). I haven't merge it yet. The bug is from cudnn layers, when you build with `-DUSE_CUDNN=ON`.

**Comment by ligongzheng** [comment]
Thank you for your answer . it seems like the page is not found!

**Comment by luoyetx** [comment]
Sorry, I merged the branch. The code is now in master branch.
